CenterlineMetrics/CenterlineMetrics.py

```diff
@@ -144,27 +145,55 @@ def onJumpCentredInSliceNodeCheckBox(self):
     self.logic.jumpCentredInSliceNode = self.ui.jumpCentredInSliceNodeCheckBox.checked
     
   def moveSliceView(self, value):
+    # +++, else, crash !!
+    if not self.logic.isInputModelValid():
+      return
     self.logic.moveSliceView(self.ui.sliceViewSelector.currentNode(), value)
     self.updateUIWithMetrics(value)
+    self.updateSurfaceAreaUIWithMetrics(value)
   
   def updateUIWithMetrics(self, value):
     tableNode = self.logic.outputTableNode
     if tableNode is None:
         return
     coordinateArray = self.logic.getRASCoordinatesAtIndex(value)
-    distance = self.logic.getUnitNodeDisplayString(tableNode.GetTable().GetValue(int(value), 0).ToDouble())
-    diameter = self.logic.getUnitNodeDisplayString(tableNode.GetTable().GetValue(int(value), 1).ToDouble())
+    distance = self.logic.getUnitNodeDisplayString(tableNode.GetTable().GetValue(int(value), 0).ToDouble(), "length")
+    diameter = self.logic.getUnitNodeDisplayString(tableNode.GetTable().GetValue(int(value), 1).ToDouble(), "length")
     coordinate = "R " + str(round(coordinateArray[0], 1))
     coordinate += ", A " + str(round(coordinateArray[1], 1))
     coordinate += ", S" + str(round(coordinateArray[2], 1))
     self.ui.distanceValueLabel.setText(distance)
     self.ui.diameterValueLabel.setText(diameter)
     self.ui.coordinatesValueLabel.setText(coordinate)
+  
+  # Show the surface area. In addition, derive the diameter of the circle having this area. Show also the difference and the ratio to the VMTK diameter.
```

Comment:
Extraction of the cross-section might not be always trivial. To allow the user to check what exact shape was used for computing the cross-sections, it would be nice to have an advanced option to generate a model that contains all the cross-sections.

Reply:
> generate a model that contains all the cross-sections

MarkupsToModel, fed with the contour points of all cross-sections, will not fit the bill, as it used vtkDelaunay3D and this is bad for concave surfaces.

I've searched a little in VTK docs and could not find a triangulator/filter that would build a 3D surface at all. May be you could hint at one tool, or elaborate your view.

Because this would amount to recreating a part of the segment, and I'm wondering about the exact usefulness.

We can always create as many static cross-section models as there are points in the centerline, that would make a confusing UI. I don't think that's what you mean though.

Using 'Combine models' ? Would it interpolate nicely between successive sections ?

Waiting for your input.

Reply:
Your can simply append each cross-section (that you use for area computation) into a single vtkPolyData using vtkAppendPolyData filter.

Reply:
Using : 

self.appendedSections.AddInputData(islandPolyData)
self.appendedSections.Update()

on the first few points, the result is as a discontinued pile of chips : 

![Screenshot](https://user-images.githubusercontent.com/33979252/125176162-a3f3a400-e1d1-11eb-9f7a-77873c8991e8.png)

Not very appealing on first view. Is it OK for you ? Or show we offer the possibility to stack sections at points chosen by the user only, and not all sections ? Once in vtkAppendPolyData, it won't be removable.

Reply:
This is good, it already allows you to spot any errors in the cross-section extraction. If you want to allow the user to see only one or a selected range then you can assign scalars to points (the same value for every point of a slice, the value can be the slice index). Then you can use Thresholding feature in Models module's Scalars section to choose which ones to display.

Reply:
OK, I see what you mean. I'll venture into that ASAP (other obligations appeared, then summer holidays).

Reply:
If you commit your latest code then I can finalize and merge it, just let me know when you are done.

Reply:
Hopefully, I should be able to finalize it in 2-3 days.

Reply:
I should be able to push a commit within 24 H.

Reply:
No problem at all, 2-3 days is completely fine, take your time. (I though I would jump in if you were to put this on hold for several weeks).

Reply:
@lassoan 
Please review my last commit.
I don't plan to add more functions for now, but have a few days to fix what you deem necessary.
Thanks.

Reply:
Hi @lassoan ,

Could you review the last commit ? If you are on summer vacation, there's nothing compelling right away.

Regards.